geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java

```diff
@@ -396,6 +398,15 @@ void doClear(RegionEventImpl regionEvent, boolean cacheWrite) {
       // Force all primary buckets to be created before clear.
       assignAllPrimaryBuckets();
 
+      for (AsyncEventQueue asyncEventQueue : partitionedRegion.getCache()
+          .getAsyncEventQueues(false)) {
+        if (((AsyncEventQueueImpl) asyncEventQueue).isPartitionedRegionClearUnsupported()) {
+          throw new UnsupportedOperationException(
+              "PartitionedRegion clear is not supported on region "
```

Comment:
I would remove "PartitionedRegion" from this message since that is more of an implementation detail. For the user they just need to know the op "clear" the region (and you added the name) and the reason (lucene index).